scalableminds/webknossos
# Auto save changes in sharing modal

This PR automatically saves settings / content of a modal. Therefore it is no longer required to explicitly hit "Save". As already stated in [#5074](https://github.com/scalableminds/webknossos/issues/5074), this makes sense for the sharing modal. Another modal is not mentioned, and I could not find another one where it makes sense to me to save the current input automatically.

The original issue #5074 mentioned checking all modals to whether auto-saving is useful in that case. On a rough look, I could not find a modal that would benefit from auto-saving (search the codebase for all files having modal in their name).
If you can think of another modal, just tell me and I'll add it here.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Open an explorative tracing
- Open the sharing modal via the drop-down menu in the navbar
- Update the sharing settings as you want (switch from private to public and set some teams and more)
- The changes in the modal should automatically be send to the backend and therefore automatically updated.

### Issues:
- fixes [#5074](https://github.com/scalableminds/webknossos/issues/5074)

------
(Please delete unneeded items, merge only when none are left open)
